specification/refarch.adoc

```diff
@@ -3,142 +3,184 @@
-for the TVM life cycle. The description in this section refers to the reference 
+We describe the capabilities of the platform to support memory isolation
+requirements for confidentiality of workloads in TVMs. We then describe
+the properties of the TSM, its instantiation, isolation and operational model
+for the TVM life cycle. The description in this section refers to the reference
 architecture in Figure 1.
 
 === CoVE Memory Isolation
 
-Memory isolation for TVMs are orchestrated by the TSM in two phases, 
-the conversion of memory to Confidential memory and the assignment of 
-confidential memory (and the enforcement of properties on its use) to TVMs. 
-Thus, CoVE requires new Physical Memory Attributes (PMAs): *Confidential* 
-and *Non-Confidential* (These are dynamic/programmable memory attributes 
-[priv ISA]). A TVM needs to access both types of memory:
-
-* Confidential memory - has Confidential PMA - used for TVM code, data
-* Non-Confidential memory - has Non-Confidential PMA - used for communication between TVM and untrusted host entities
-
-The COVH ABIs implemented by the TSM provides interfaces to the VMM to convert / 
-donate memory to Confidential [Convert] and vice-versa [Reclaim]. 
-TVM memory is by default assigned from Confidential memory regions. 
-TVM may be assigned shared memory regions. Both properties are enforced 
-by the TSM. Figure 3 below shows the abstract model of isolation:
-
-* Hart with Confidential mode qualifier =1 is allowed to access Confidential 
-and Non-Confidential memory
-* Hart with Confidential mode qualifier =0 is allowed to access only Non-confidential Memory
+Memory isolation for TVMs is orchestrated by the TSM-driver and the TSM in two
+phases: the conversion of memory to confidential memory and the assignment of
+confidential memory (alongwith the enforcement of properties on use) to TVMs.
+To enforce isolation across Host and Confidential supervisor domains, CoVE
+requires page-based isolation of physical memory. There are two deployment
+models possible here. CoVE ABI is equally applicable for both modes - this
+specification focusses on the second deployment model (b) where a peer
+supervisor domain is used to host confidential workloads.
+
+a. When the TSM is the only root HS mode component on the platform, the G-stage
+page table can be used to enforce isolation between confidential TVMs and
+ordinary VMs. In this model the host VMM must execute in the de-privileged VS
+mode and the TSM must provide nested virtualization of the H-extension controls.
+This model may be suitable for client/embedded systems.
+
+b. The TSM operates in S/HS mode as a peer supervisor domain manager to the
+hosting supervisor domain which operates in S/HS mode as well. This model uses
+the MTT along with Sv and G-stage PT for confidential TVM isolation. The MTT
+is used to assign physical memory to the Confidential supervisor domain called
```

Comment:
Sv is not used for TVM isolation. On G-stage.

Reply:
meant to say Sv may be used by the guest OS as normal (updated text)